src/main/java/com/github/jcustenborder/kafka/connect/transform/common/HeaderToField.java

```diff
@@ -100,6 +101,32 @@ Conversion conversion(Schema schema) {
     });
   }
 
+  @Override
+  protected SchemaAndValue processMap(R record, Map<String, Object> input) {
+    if (record.headers().isEmpty()) {
+      return new SchemaAndValue(null, input);
+    }
+
+    Map<String, Object> headers = new HashMap<>();
+    if (this.config.mappings.isEmpty()) {
+      for (Header header: record.headers()) {
+        headers.put(header.key(), header.value());
+        break;
+      }
+    } else {
+      this.config.mappings.forEach(mapping -> {
+        for (Header header: record.headers()) {
+          if (header.key().equals(mapping.header)) {
+            headers.put(mapping.field, header.value());
+            break;
+          }
+        }
+      });
+    }
+
+    input.put("_headers", headers);
```

Comment:
I just tested this code because we are running into a similar problem.
It appears to add the extracted values to the new struct field `_headers` which you are setting here, creating a nested structure.
So
```
".header.mappings": "time:INT64:d,time:INT64:h,time:INT64:m"
```
becomes 
```
"value": {
    "_headers" : {
    "d" : 1669852804800000000,
    "h" : 1669852804800000000,
    "m" : 1669852804800000000
    }
}
```
I would have expected a flat structure here (and interestingly also the test you added shows a flat structure).

Reply:
The test actually doesn't test this correctly.
To have the same behaviour as for Structs (i,e, adding the new fields to the root), this should be:

```suggestion
    input.putAll(headers);
```